### ISSUE TITLE
Clarification on HEIF (23008-12) about ignored descriptive properties

### ISSUE THREAD

**Issue by y-guyon** [MPEGGroup/FileFormat#111]
Section 6.5.1 of HEIF says:

> Readers shall allow and ignore descriptive properties following the first transformative or unrecognized property, whichever is earlier, in the sequence associating properties with an item.

- What is an unrecognized property?
- What is the effect of ignoring a descriptive property? Does it happen at the library or application level?

See https://github.com/AOMediaCodec/libavif/issues/2525 for context and examples.

**Comment by mhannuksela** [comment]
Hi Yannis,

Here are my opinions on your questions.

> What is an unrecognized property?

A property that a reader does not recognize. This could be spelled out in the cited sentence.

> What is the effect of ignoring a descriptive property?

None. The cited requirement ensures that a reader does not misinterpret a descriptive property to describe the image before transformations.

> Does it happen at the library or application level?

This is an implementation question, outside of the scope of the HEIF standard.

**Comment by leo-barnes** [comment]
>> What is an unrecognized property?

> A property that a reader does not recognize. This could be spelled out in the cited sentence.

I agree with this interpretation as well. Worth adding is that certain brands mandate that parsers complying with that brand shall understand certain item properties.

**Comment by y-guyon** [comment]
Thank you for the explanations.

In the context of libavif I think it is fine:

- libavif recognizes some descriptive properties itself
- libavif forwards clap, irot and imir through `avifImage` to the user (libavif does not apply them itself)
- libavif discards the whole item if an unrecognized property is essential, and transformative properties must be marked as such
- libavif forwards all other unrecognized properties through `avifImage` to the user

So libavif does not ignore any property but lets the user handle that.

**Comment by y-guyon** [comment]
Follow-up at https://github.com/MPEGGroup/FileFormat/issues/113.
